repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -385,42 +387,25 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         wrappedRequest = new HttpServletRequestWrapper((HttpServletRequest) request, bufferedInputStream);
 
-        // OpenTracing - set up scope placeholder.
-        Scope scope = null;
-
-        if (openTracingService.isPresent() && openTracingService.get().isEnabled()) {
-            LOG.trace("We enabled OpenTracing service.  Let's see if there are any passed-in spans");
-
-            SpanContext context = null;
-
-            try {
-                context = openTracingService.get().getGlobalTracer().extract(
-                    Format.Builtin.HTTP_HEADERS, new TracerExtractor(wrappedRequest));
-            } catch (RuntimeException re) {
-                LOG.error("Incoming tracer could not be parsed.  We're going to start a new root span here; even though" +
-                    "this is most likely part of a larger span.  Check out thrown exception for more details", re);
-            }
-
-            LOG.debug("Got the span context from request: {}", context);
-
-            Span activeSpan;
-
-            if (context == null)
-                activeSpan = openTracingService.get().getGlobalTracer()
-                    .buildSpan(String.format("%s %s", wrappedRequest.getMethod(),
-                        wrappedRequest.getRequestURI()))
-                    .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT).start();
-            else
-                activeSpan = openTracingService.get().getGlobalTracer()
-                    .buildSpan(String.format("%s %s", wrappedRequest.getMethod(),
-                        wrappedRequest.getRequestURI()))
-                    .asChildOf(context)
-                    .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT).start();
-
-            scope = openTracingService.get().getGlobalTracer().scopeManager().activate(activeSpan, false);
+        LOG.trace("Let's see if there are any OpenTracing spans passed-in");
+        SpanContext context = null;
+        try {
+            context = tracer.extract(Format.Builtin.HTTP_HEADERS, new TracerExtractor(wrappedRequest));
+        } catch (RuntimeException re) {
+            LOG.error("Incoming tracer could not be parsed.");
+            LOG.error("We're going to start a new root span here; even though this is most likely part of a larger span.");
+            LOG.error("Check out following exception for more details:", re);
+        }
+        LOG.debug("Got the span context from request: {}", context);
 
-            LOG.debug("Start a new span {}", scope.span());
+        Tracer.SpanBuilder spanBuilder = tracer.buildSpan(String.format("%s %s", wrappedRequest.getMethod(), wrappedRequest.getRequestURI()));
+        if (context != null) {
+            spanBuilder = spanBuilder.asChildOf(context);
         }
+        Span activeSpan = spanBuilder.withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT).start();
+
+        Scope scope = tracer.scopeManager().activate(activeSpan, false);
```

Comment:
Since you active the span right after creating it, and `activeSpan` is not referenced later, why not just use `startActive(...)` rather than `start()`?

Reply:
Also, since every time the `Scope` is `finished()` it is also `closed()`, why not call `startActive(true)` rather than `startActive(false)`? That is, auto-finish the `Scope` on `close()`, and then no need to call `finish()`.

Reply:
Sure. I definitely don't have the familiarity with this API yet and that is basically what was in the original prototype cleaned up a bit. All that is now in place.